File: tests/e2e/scripts/error_handling.ts
function riskyOperation(x: number): number {
    try {
        if (x < 0) {
            throw new Error("negative");
        }
        console.log("success:", x);
    } catch (e) {
        console.log("caught error");
    }
    return x;
}

// try/catch 両方が return する関数 
function safeDivide(a: number, b: number): number {
    try {
        if (b === 0) {
            throw new Error("div by zero");
        }
        return a / b;
    } catch (e) {
        return 0;
    }
}

function main(): void {
    riskyOperation(5);
    riskyOperation(-1);
    console.log("safe divide 10/2:", safeDivide(10, 2));
    console.log("safe divide 10/0:", safeDivide(10, 0));
}

File: tests/e2e/scripts/error_handling.rs
fn riskyOperation(x: f64) -> f64 {
    let mut _try_result: Result<(), String> = Ok(());
    'try_block: {
        if x < 0.0 {
            _try_result = Err("negative".to_string());
            break 'try_block;
        }
        println!("{} {}", "success:", x);
    }
    if let Err(e) = _try_result {
        println!("caught error");
    }
    x
}

fn safeDivide(a: f64, b: f64) -> f64 {
    let mut _try_result: Result<(), String> = Ok(());
    'try_block: {
        if b == 0.0 {
            _try_result = Err("div by zero".to_string());
            break 'try_block;
        }
        return a / b;
    }
    if let Err(e) = _try_result {
        return 0.0;
    }
    unreachable!();
}

fn main() {
    riskyOperation(5.0);
    riskyOperation(-1.0);
    println!("{} {}", "safe divide 10/2:", safeDivide(10.0, 2.0));
    println!("{} {}", "safe divide 10/0:", safeDivide(10.0, 0.0));
}
